pkg/operation/botanist/controlplane.go

```diff
@@ -533,3 +499,43 @@ func (b *Botanist) DeployGardenerResourceManager(ctx context.Context) error {
 
 	return b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-controlplane", "charts", name), b.Shoot.SeedNamespace, name, values, nil)
 }
+
+// DeployBackupEntryInGarden deploys the BackupEntry resource in garden.
+func (b *Botanist) DeployBackupEntryInGarden(ctx context.Context) error {
+	var (
+		name        = common.GenerateBackupEntryName(b.Shoot.Info.Status.TechnicalID, b.Shoot.Info.Status.UID)
+		backupEntry = &gardencorev1alpha1.BackupEntry{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: b.Shoot.Info.Namespace,
+			},
+		}
+		bucketName string
+		seedName   *string
+	)
+
+	if err := b.K8sGardenClient.Client().Get(ctx, kutil.Key(b.Shoot.Info.Namespace, name), backupEntry); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		// If backupEntry doesn't already exists, we have to assign backupBucket to backupEntry.
+		if b.Seed.Info.Spec.Backup.Region != nil && len(*b.Seed.Info.Spec.Backup.Region) != 0 {
```

Comment:
```suggestion
		bucketName = common.GenerateBackupBucketName(b.Seed.Info.Name, b.Seed.Info.Spec.Cloud.Region, b.Seed.Info.UID)
		if b.Seed.Info.Spec.Backup.Region != nil && len(*b.Seed.Info.Spec.Backup.Region) != 0 {
```

saves the else branch